var/spack/repos/builtin/packages/mumps/package.py

```diff
@@ -125,28 +128,49 @@ def write_makefile_inc(self):
         fpic = '-fPIC' if '+shared' in self.spec else ''
         # TODO: test this part, it needs a full blas, scalapack and
         # partitionning environment with 64bit integers
+
         if '+int64' in self.spec:
-            makefile_conf.extend(
-                # the fortran compilation flags most probably are
-                # working only for intel and gnu compilers this is
-                # perhaps something the compiler should provide
-                ['OPTF    = %s -O  -DALLOW_NON_INIT %s' % (fpic, '-fdefault-integer-8' if self.compiler.name == "gcc" else '-i8'),  # noqa
-                 'OPTL    = %s -O ' % fpic,
-                 'OPTC    = %s -O -DINTSIZE64' % fpic])
+            if self.compiler.name == "xl" or self.compiler.name == "xl_r":
+                makefile_conf.extend(
+                    ['OPTF    = -O3',
+                     'OPTL    = %s -O3' % fpic,
+                     'OPTC    = -O3'])
+            else:
+                makefile_conf.extend(
+                    # the fortran compilation flags most probably are
+                    # working only for intel and gnu compilers this is
+                    # perhaps something the compiler should provide
+                    ['OPTF    = %s -O  -DALLOW_NON_INIT %s' % (fpic, '-fdefault-integer-8' if self.compiler.name == "gcc" else '-i8'),  # noqa
+                     'OPTL    = %s -O ' % fpic,
+                     'OPTC    = %s -O -DINTSIZE64' % fpic])
         else:
-            makefile_conf.extend(
-                ['OPTF    = %s -O  -DALLOW_NON_INIT' % fpic,
-                 'OPTL    = %s -O ' % fpic,
-                 'OPTC    = %s -O ' % fpic])
+            if self.compiler.name == "xl" or self.compiler.name == "xl_r":
+                makefile_conf.extend(
+                    ['OPTF    = -O3',
+                     'OPTL    = %s -O3' % fpic,
+                     'OPTC    = -O3'])
+            else:
+                makefile_conf.extend(
+                    ['OPTF    = %s -O  -DALLOW_NON_INIT' % fpic,
+                     'OPTL    = %s -O ' % fpic,
+                     'OPTC    = %s -O ' % fpic])
 
         if '+mpi' in self.spec:
             scalapack = self.spec['scalapack'].scalapack_libs
-            makefile_conf.extend(
-                ["CC = %s" % join_path(self.spec['mpi'].prefix.bin, 'mpicc'),
-                 "FC = %s" % join_path(self.spec['mpi'].prefix.bin, 'mpif90'),
-                 "FL = %s" % join_path(self.spec['mpi'].prefix.bin, 'mpif90'),
-                 "SCALAP = %s" % scalapack.ld_flags,
-                 "MUMPS_TYPE = par"])
+            if (self.spec.satisfies('%xl') or self.spec.satisfies('%xl_r')) and self.spec.satisfies('^ibm-mpi'): # noqa
+                makefile_conf.extend(
+                    ["CC = %s" % join_path(self.spec['mpi'].prefix.bin, 'mpixlc'), # noqa
```

Comment:
i think this is wrong way to do it. Just use `self.spec['mpi'].mpicc` and alike. `ibm-mpi` should set these correctly. It's the responsibility of MPI package to do not where you use it. Otherwise you would be adding these lines in every single package which needs to compile with `mpi` wrappers. In other words, you should be able to workaround without `if-else` condition here.